### PR TITLE
Build signup page 

### PR DESCRIPTION
**Pull request by muath-khatib**
Relates #10 

1. Build the sign-up page with client-side validation and add some transition.
2. Remove the Header component from the App file and set it on the Home page component.

### CODE REVIEW

**Review by emanNahed — DISMISSED** [review]

File: client/src/components/SignUp/SignUpForm.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    },
+  },
+};
+
+const SignUpForm = () => {
+  const [signupData, setSignupData] = useState({});
+
+  useEffect(() => {
+    console.log(signupData);
+    return () => console.log('clean');
```
Comment (Mu7ammadAbed) [inline client/src/components/SignUp/SignUpForm.jsx:28]: !!!

File: client/src/components/SignUp/index.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    setInProp(true);
+
+    return () => setInProp(false);
+  }, []);
+
+  return (
+    <CSSTransition in={inProp} timeout={1000} classNames="css-transition">
+      <section className="signup-section">
+        <section className="form-section">
+          <h3>Sign up with REMAN</h3>
```
Comment (Mu7ammadAbed) [inline client/src/components/SignUp/index.jsx:22]: Use `antd` `Title` component

File: client/src/components/SignUp/index.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        <Social />
+          <div className="form-container">
+            <SignUpForm />
+          </div>
+        </section>
+        <section className="bg-section">
+          <img className="bg-img" src={signup} alt="signup-bg" />
```
Comment (Mu7ammadAbed) [inline client/src/components/SignUp/index.jsx:29]: Convert `img` into a common component

File: client/src/pages/Home.jsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import Header from '../components/Header';
+
+const Home = () => (
+  <>
+    <Header />
+  </>
```
Comment (Mu7ammadAbed) [inline client/src/pages/Home.jsx:6]:
```suggestion
    <Header />
```

File: package.json [hunk 5]
```diff
@@ -20,7 +20,8 @@
     "express": "^4.17.1",
     "joi": "^17.4.2",
     "jsonwebtoken": "^8.5.1",
-    "pg": "^8.7.1"
+    "pg": "^8.7.1",
+    "react-transition-group": "^4.4.2"
```
Comment (Mu7ammadAbed) [inline package.json:24]: why you installed this here ?

File: client/src/components/SignUp/SignUpForm.jsx (hunk 1 above)
Comment (muath-khatib) [reply client/src/components/SignUp/SignUpForm.jsx:28]: There is no database and routes to work on it, I will fetching process when the back-side is ready 
the console is just for testing the outputs

File: package.json (hunk 5 above)
Comment (muath-khatib) [reply package.json:24]: It's not installing, after pulling the main branch  and the ( "react-transition-group": "^4.4.2") module added there is a ( , ) added in the package Json file

**Review by emanNahed — APPROVED** [review]

**Review by reemsh2000 — APPROVED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by aaamra — APPROVED** [review]

**Pull request merged**